### PR TITLE
Feature/data structure

### PR DESCRIPTION
**Pull request by jamieadkins95**
I've amended the gwentify script to be as close to the data structure we want with the info they provide.

Feel free to discuss anything I have changed, I've commented on the major stuff below.

(Also I didn't commit any changes to latest.json, keeps the PR tidy)

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def run(self):
                 key = getNameKey(cardData['name'])
                 cardData['key'] = key
                 self.finalDataQueue.put(cardData)
-                self.imageQueue.put((cardData['key'], cardData['imageUrl']))
+                self.imageQueue.put((cardData['key'], cardData['variations'][0]['art']['fullsizeImageUrl']))
```
Comment (jamieadkins95) [inline __init__.py:107]: I've introduced the notion of variations, of which each can have their own artwork.

Gwentify will only ever have one variation, so it's almost redundant here, but it is part of our final goal.

File: gwentifyHandler.py [hunk 2]
```diff
@@ -82,7 +82,17 @@ def getCardJson(html):
...
+
+    # Only one variation in gwentify, but the game can have many variations of a card
+    variation = {}
+    variation["availability"] = "BaseSet" # Currently all cards are from the base set.
+    art = {}
+    art["fullsizeImageUrl"] = imageUrl.strip()
+
+    variation["art"] = art
+
+    dataMap["variations"].append(variation)
```
Comment (jamieadkins95) [inline gwentifyHandler.py:95]: Introduced the variations from [here](https://gist.github.com/jamieadkins95/b3f23961d5260e26b2798fa6240a9f0e).

File: gwentifyHandler.py [hunk 3]
```diff
@@ -92,14 +102,16 @@ def getCardJson(html):
 
         # We check the name and do the appropriate action to store it in the map.
         if attribute == "Group:":
-            dataMap["type:"] = data.a.get_text().replace(u'\u00a0', u' ').strip()
+            dataMap["type"] = data.a.get_text().replace(u'\u00a0', u' ').strip()
```
Comment (jamieadkins95) [inline gwentifyHandler.py:105]: I think these were typos? I assume you don't want the ":" to be part of the key.

File: gwentifyHandler.py [hunk 4]
```diff
@@ -165,10 +177,9 @@ def getCardJson(html):
     # Same as for info.
     try:
         flavor = cardArticle.find('p', class_="flavor").get_text().replace(u'\u00a0', u' ').strip()
-        # According to the format agreed with Chris and Jamie A., we will name it "flavour" and not "flavor".
-        dataMap["flavour"] = flavor
+        dataMap["flavor"] = flavor
```
Comment (jamieadkins95) [inline gwentifyHandler.py:180]: These is completely my fault, we should be using the American spelling, my brain just defaulted to the British spelling.

**Review by jamieadkins95 — COMMENTED** [review]
Everything I've changed is up for discussion here, what do you think?

**Comment by Tri125** [comment]
Thank you very much for your PR. Good catch on the colon, I didn't notice.

**Pull request merged**